lib/Conversion/AIEVecToLLVM/AIEVecToLLVM.cpp

```diff
@@ -1257,16 +1266,17 @@ class BroadcastScalarOpConversion
       unsigned srcBitWidth = srcType.getIntOrFloatBitWidth();
 
       if (srcBitWidth < 32) {
-        src = rewriter.create<LLVM::SExtOp>(loc, rewriter.getI32Type(),
-                                            adaptor.getSource());
+        src = rewriter.create<LLVM::SExtOp>(loc, rewriter.getI32Type(), src);
       } else if (srcBitWidth > 32) {
-        src = rewriter.create<LLVM::TruncOp>(loc, rewriter.getI32Type(),
-                                             adaptor.getSource());
+        src = rewriter.create<LLVM::TruncOp>(loc, rewriter.getI32Type(), src);
```

Comment:
What are the valid use cases where the source of the broadcast has bitwidth > 32 and truncating it to 32 bits be preserve semantics?

Reply:
Nice catch. We have limited the source scalar to be in `BF16, F32, I32, I16, I8` types. I will remove this line.